src/System.Text.Json/src/System/Text/Json/JsonElement.cs

```diff
@@ -0,0 +1,372 @@
+
+            // This method is try, but calls a non-try, because in the future
+            // JsonDocument and JsonElement may support non-contiguous and/or
+            // differently-encoded data.  JsonDocument's version of this method
+            // would likewise become Try when that happens.
+            rawValue = _parent.GetRawValue(_idx);
+            return true;
+        }
+
+        /// <summary>
+        ///   Attempt to copy the uninterpreted memory contributing to the value of this element
+        ///   into <paramref name="destination"/> as UTF-8 code units.
+        /// </summary>
+        /// <param name="destination">
+        ///   Buffer into which the uninterpreted value should be copied.
+        /// </param>
+        /// <param name="bytesWritten">
+        ///   Receives the number of bytes written to <paramref name="destination"/>.
+        /// </param>
+        /// <returns>
+        ///   <c>true</c> if <paramref name="destination"/> was big enough to receive the UTF-8
+        ///   version of the uninterpreted element value, <c>false</c> otherwise.
+        /// </returns>
+        public bool TryCopyRawValue(Span<byte> destination, out int bytesWritten)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            // If non-contiguous memory is supported by JsonDocument, this method should determine
+            // the total byte count and then write to destination.
+            //
+            // If non-UTF-8 encodings are supported by JsonDocument, this method should transcode.
+
+            ReadOnlyMemory<byte> rawData = _parent.GetRawValue(_idx);
+
+            if (rawData.Length > destination.Length)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            rawData.Span.CopyTo(destination);
+            bytesWritten = rawData.Length;
+            return true;
+        }
+
+        public bool GetBoolean()
+        {
+            JsonTokenType type = Type;
+
+            return
+                type == JsonTokenType.True ? true :
+                type == JsonTokenType.False ? false :
+                throw new InvalidOperationException();
+        }
+
+        public string GetString()
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.GetString(_idx);
+        }
+
+        public bool TryGetValue(out int value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public int GetInt32()
+        {
+            if (TryGetValue(out int value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetValue(out long value)
```

Comment:
I think this was discussed in the API review, but seeing it written out now, it seems strange that there's GetInt32 and TryGetValue, rather than GetInt32 and TryGetInt32.  I understand that it's needed for the Get and not needed for the TryGet because of overload resolution on the out argument, but it still feels inconsistent.  I think it'd be better as GetInt32 and TryGetInt32.